pkg/install/0-installstorage.go

```diff
@@ -37,7 +38,11 @@ var apiVersions = map[string]string{
 }
 
 func (i *Installer) createDNS(ctx context.Context) error {
-	return i.dns.Create(ctx, i.doc.OpenShiftCluster)
+	err := i.dns.Create(ctx, i.doc.OpenShiftCluster)
+	if err != nil {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeDNSCollision, "properties.clusterProfile.domain", "DNS creation failed with error %s.", err)
```

Comment:
leave target empty like we do in dynamic validation

Reply:
message can be something like `"The provided domain '%s' is already in use by a cluster."`